includes/class-wc-stripe-api.php

```diff
@@ -242,6 +263,13 @@ public static function retrieve( $api ) {
 			]
 		);
 
+		// If we get a 401 error, we know the secret key is not valid, we save a transient to avoid making calls until the secrect key gets updated.
+		if ( ! empty( $response['response']['code'] ) && 401 === $response['response']['code'] ) {
```

Comment:
We should guard against `$response` being a `WP_Error` instance.
```suggestion
		if ( is_array( $response ) && ! empty( $response['response']['code'] ) && 401 === $response['response']['code'] ) {
```